src/fu-engine.c

```diff
@@ -222,6 +222,78 @@ fu_engine_watch_device (FuEngine *self, FuDevice *device)
 			  G_CALLBACK (fu_engine_status_notify_cb), self);
 }
 
+static gboolean
+fu_engine_update_motd (FuEngine *self, GError **error)
```

Comment:
I think this should be split out in `fu-engine-common.c` or `fu-engine-helper.c` or something, as `fu-engine.c` is pretty big now, and all this is self-contained.